graph_client:
  ros__parameters:
    # Subscribers.
    opt_graph_topic: /graph_monitor/sparse_graph/graph
    opt_traj_topic: /graph_monitor/sparse_graph/trajectory
    est_traj_topic: /trajectory
    est_traj_path_topic: /optimized_path
    opt_submap_constraint_topic: /graph_monitor/submaps

    # Publishers.
    anchor_node_topic: "/graph_client/anchor_nodes"
    relative_node_topic: "/graph_client/relative_nodes"
    intra_constraints: "/graph_client/relative_nodes"
    verification_service: "/graph_monitor/verification"

    # Subscriber and Publisher
    client_update_topic: "/graph_client/latest_graph"

    # Constraint construction:     multiscale, euclidean, always, absolute
    client_mode: "multiscale"
    classifier: "top" # "top" or "simple"
    top_classifier_select_n: 10
    large_scale_partition_method: "id" # 'id' or 'nth'
    n_hop_mid_constraints: 5
    min_dist_large_constraints: 20.0
    stop_method: "" # "", "dirichlet", "tv", "alv"
    stop_threshold: 0.1

    # Graph configuration.
    construction_method: "so3" # "se3" or "so3" or "r3"
    use_graph_hierarchies: False
    max_graph_levels: 4 # 2 means 1 level of hierarchy
    graph_hierarchies_node_threshold: 1
    use_parallel_construction: False

    # Div
    dataroot: "/home/berlukas/Documents/workspace/ros2/fgsp_ws/src/fgsp"
    robot_name: "cerberus" # cerberus, euroc_X, anymal_X
    update_rate: 0.05
    enable_client_update: False
    enable_anchor_constraints: False
    enable_relative_constraints: True
    enable_submap_constraints: True
    enable_signal_recording: True
    enable_trajectory_recording: True
    signal_export_path: "/data/{src}_signal.npy"
    graph_coords_export_path: "/data/{src}_graph_coords.npy"
    graph_adj_export_path: "/data/{src}_graph_adj.npy"
    trajectory_export_path: "/data/{src}_trajectory.npy"
    trajectory_raw_export_path: "/data/{src}_raw_trajectory.npy"
    label_output_path: "/data/opt_labels.dat"
    degenerate_window: 20
    synchronization_max_diff_s: 0.5
    verbosity: 7
    warmup_nodes: 190
    max_iterations: 1

    T_robot_server:
      [
        0.03547377,
        0.99931998,
        0.01005935,
        0.17368547,
        -0.9993442,
        0.03539785,
        0.00762637,
        -2.14932286,
        0.00726511,
        -0.0103233,
        0.99992032,
        -0.61567595,
        0.,
        0.,
        0.,
        1.,
      ]
